sql-core/src/main/java/io/cdap/sql/core/SQLFilter.java

```diff
@@ -0,0 +1,81 @@
+package io.cdap.sql.core;
+
+import io.cdap.sql.core.enums.SQLPredicateOperator;
+import io.cdap.sql.core.interfaces.Operand;
+
+public class SQLFilter extends ExpressionTreeNode<Operand, SQLPredicateOperator> {
+  private SQLFilter(Operand left, Operand right, SQLPredicateOperator operator) {
+    super(null, null, left, right, operator);
+  }
+
+  private SQLFilter(SQLFilter left, SQLFilter right, SQLPredicateOperator operator) {
+    super(left, right, null, null, operator);
+  }
+
+  public SQLFilter getLeftFilter() {
+    return (SQLFilter) super.getLeftNode();
+  }
+
+  public SQLFilter getRightFilter() {
+    return (SQLFilter) super.getRightNode();
+  }
+
+  public static Builder builder() { return new Builder(); }
+
+  public static class Builder {
+    private SQLFilter leftNode;
+    private SQLFilter rightNode;
+    private Operand leftOperand;
+    private Operand rightOperand;
+    private SQLPredicateOperator operation;
+
+    private Builder() {}
+
+    public Builder setLeftNode(SQLFilter node) {
+      this.leftNode = node;
+      return this;
+    }
+
+    public Builder setRightNode(SQLFilter node) {
+      this.rightNode = node;
+      return this;
+    }
+
+    public Builder setLeftOperand(Operand op) {
+      this.leftOperand = op;
+      return this;
+    }
+
+    public Builder setRightOperand(Operand op) {
+      this.rightOperand = op;
+      return this;
+    }
+
+    public Builder setOperation(SQLPredicateOperator op) {
+      this.operation = op;
+      return this;
+    }
+
+    public SQLFilter build() {
+      if (leftNode != null && rightNode != null) {
```

Comment:
Yeah I think you're right here, I changed it to a builder since I thought it'd be easier to read an filter expression tree if it was created using a builder, but the checks become really weird to enforce.